dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorNodeSelectionStrategyChannel.java

```diff
@@ -131,28 +131,33 @@ LimitedChannel getCurrentChannel() {
         int pin = currentPin.get();
         PinChannel channel = nodeList.get(pin);
```

Comment:
If we want to take this another step, the only state the FutureCallback requires is the pin, so we could allocate one callback for each PinChannel up front and reuse them.

Reply:
I _almost_ did this, but I think the upside is gonna be pretty marginal compared for quite a big readability cost - all of these optimizations are dwarfed by how slow the instrumented channel is

Reply:
I haven't actually found the instrumented channel helpful, it's too broad to provide per-endpoint information, and not helpful from a high level because it's almost entirely dependent upon the server handling.
What if we removed the timer? Alternatively what if we removed the service-name tag from the timer? Either way we can cut dd spend and I don't think we lose any actionable data.